### PR TITLE
SQL: use calendar interval of 1y instead of fixed interval for grouping by YEAR and HISTOGRAMs

### LINKED ISSUE #40162: SQL: inconsistent results for YEAR method
For the following test data:

```
{"index":{}}
{"certified":"2019-02-21T10:44:11.857Z"}
{"index":{}}
{"certified":"2018-11-30T10:44:11.857Z"}
{"index":{}}
{"certified":"2019-01-14T10:44:11.857Z"}
{"index":{}}
{"certified":"2019-03-15T10:44:11.857Z"}
{"index":{}}
{"certified":"2018-07-31T10:44:11.857Z"}
{"index":{}}
{"certified":"2018-08-21T15:44:11.857Z"}
{"index":{}}
{"certified":"2019-03-11T15:44:11.857Z"}
```

and query `SELECT count(*), YEAR(certified) year FROM certification WHERE certified > 0 GROUP BY year` we get:

```
   count(*)    |     year      
---------------+---------------
3              |2017           
4              |2018           
```

But for a slightly modified query (by adding another grouping): `SELECT count(*), YEAR(certified) year FROM certification WHERE certified > 0 GROUP BY year, certified` we get:

```
   count(*)    |     year      
---------------+---------------
1              |2018           
1              |2018           
1              |2018           
1              |2019           
1              |2019           
1              |2019           
1              |2019           
```

### PR DESCRIPTION
**Pull request by astefan**
Because Elasticsearch forces users to use a fixed interval for all nX periods of time (3 days, 4 years, 5 months etc), to have a consistent behavior throughout SQL, the `fixed_interval` approach was chosen. But this resulted in having an [unintuitive behavior for intervals of one year](https://github.com/elastic/elasticsearch/issues/40162).
This PR changes grouping by YEAR() and HISTOGRAMs with `INTERVAL '1' YEAR` to use a `date_histogram` with `calendar_interval` of `1y`.

Fixes https://github.com/elastic/elasticsearch/issues/40162.

### CODE REVIEW

**Review by matriv — APPROVED** [review]
LGTM

File: docs/reference/sql/functions/grouping.asciidoc [hunk 1]
```diff
@@ -86,6 +86,10 @@ the multiple of a day. E.g.: for `HISTOGRAM(CAST(birth_date AS DATE), INTERVAL '
 actually used will be `INTERVAL '2' DAY`. If the interval specified is less than 1 day, e.g.:
 `HISTOGRAM(CAST(birth_date AS DATE), INTERVAL '20' HOUR)` then the interval used will be `INTERVAL '1' DAY`.
 
+[IMPORTANT]
+All intervals specified for a date/time HISTOGRAM will use a <<search-aggregations-bucket-datehistogram-aggregation,fixed interval>>
+in their `date_histogram` aggregation definition, with the notable exception of `INTERVAL '1' YEAR` where a calendar interval is used.
```
Comment (costin) [inline docs/reference/sql/functions/grouping.asciidoc]: It's worth explaining what that means regarding the start of the year vs exact time difference...

**Review by costin — APPROVED** [review]
LGTM

**Comment by costin** [comment]
Make sure to indicate the versions selected for backporting. Considering it's a behavior change, I'm on the fence on whether we should do any backporting.

**Comment by astefan** [comment]
@elasticmachine run elasticsearch-ci/packaging-sample

**Pull request merged**